### DISCUSSION TITLE
Integrity constraint violation: 1048 Column 'name' cannot be null

### DISCUSSION THREAD

**Discussion by nataliiatytovych** [Laravel-Backpack/community-forum#1051] [Q&A (Help)] [answered]
i am using laravel pro and define a column:
$this->crud->addField([
            'name' => 'name,
            'label' => __('name'),
            'type' => 'name',
            'default' => '',
        ]);
      
but after saving i catch an error  "Integrity constraint violation: 1048 Column 'name' cannot be null" and see that this column value is not set to "". I don't want to change by db columns because of lib bugs.

**Comment by karandatwani92** [comment]
Hey @nataliiatytovych 

Blank spaces are trimmed before saving, and no value turns to `NULL`.
- From the error, I can say that your DB column `name` is set as `NOT NULL`; it does not allow `NULL` value.
- if the `name` field is required, you must set validation in the Request file to stop this error. `'name' => 'required|min:5|max:255'`.
- you don't need to set `default` in case of blank.

```diff
$this->crud->addField([
   'name' => 'name',
   'label' => __('name'),
-    'type' => 'name',
+   'type' => 'text',
-   'default' => '',
]);
```

**Reply by nataliiatytovych** [reply]
thanks for the answer 

this column is not nullable in DB that's why i want to set default value "" (empty string) for it in case of empty field. But this don't work

**Reply by karandatwani92** [reply, marked as answer]
The empty string gets trimmed and becomes null. It's expected, not a bug.
You have two options:
- You change your DB column to accept null
- Or set (empty string) value conditionally before saving via Laravel model [events](https://laravel-news.com/working-with-laravel-model-events)

**Reply by nataliiatytovych** [reply]
thanks a lot!
